File: TestFirebase/app/src/main/java/com/example/testfirebase/MainActivity.java
package com.example.testfirebase;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;

public class MainActivity extends AppCompatActivity {

    // https://firebase.google.com/docs/cloud-messaging/android/client
    // https://console.firebase.google.com/project/testfirebase-1c33c/notification

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }
}

File: CurrencyExchanger/app/src/main/java/com/example/currencyexchanger/MainActivity.kt
package com.example.currencyexchanger

import android.os.Bundle
import android.util.Log
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.example.currencyexchanger.data.Repository
import com.example.currencyexchanger.data.converter.EntityConverter
import com.example.currencyexchanger.data.network.NetworkModule
import okhttp3.OkHttpClient
import okhttp3.Request
import com.google.gson.annotations.SerializedName
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch


class MainActivity : AppCompatActivity() {
    private val repo = Repository(NetworkModule(), EntityConverter())
    private lateinit var tv: TextView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        initViews()
        doCall()
    }

    private fun initViews() {
        tv = findViewById(R.id.tv)
    }

    private fun doCall() {
        CoroutineScope(Dispatchers.IO).launch {
            val result = repo.getList()
            tv.post { tv.text = result.toString() + "\n" + result.rates.toString() }
        }


    }
}